src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -411,6 +424,15 @@ public void Dispose()
                 {
                     Interlocked.Decrement(ref _fileMonitoringService._suspensionRequestsCount);
                     _fileMonitoringService._typedLogger.LogDebug($"Exiting restart suspension scope. ({_fileMonitoringService._suspensionRequestsCount} requests).");
+                    if (_autoResume)
+                    {
+                        // Suppressing execution context as it can cause scope container disposing issue
+                        using (System.Threading.ExecutionContext.SuppressFlow())
+                        {
+                            // Attempting to restart in the background
+                            Task.Run(async () => await _fileMonitoringService.RestartIfScheduledAsync());
```

Comment:
Let's just remove the RestartIfScheduledAsync method and inline the work here.

Reply:
I still think we need `RestartIfScheduledAsync` if we want to check if there was a file change which needs a restart.